### DISCUSSION TITLE
Mapping Reference Tables to TypeScript Enums?

### DISCUSSION THREAD

**Discussion by XiNiHa** [prisma/prisma#5236] [Q&A]
I have a reference table with `id` and `name`, and fields that reference ids from the table as foreign keys. Is it possible to map those as TypeScript enums? I've searched a bit on both the documentation and issues, but haven't found anything useful.

**Comment by 2color** [comment]
Hey @XiNiHa 👋🏼,

Could you share the Prisma schema to get a better understand of what you're trying to do?

**Reply by XiNiHa** [reply]
https://gist.github.com/XiNiHa/2e94e45fa8641e1cb2fbcff4a40d200b

I want to map each row of `user_role` and `user_sex` as TypeScript enums, which makes the type of `user.role` and `user.sex` to be appropriately mapped TS enums, not `Int`s.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
